core/clustersmngr/factory.go

```diff
@@ -469,6 +494,102 @@ func (cf *clustersManager) userNsList(ctx context.Context, user *auth.UserPrinci
 	return cf.GetUserNamespaces(user)
 }
 
+func (cf *clustersManager) getOrCreateClient(ctx context.Context, user *auth.UserPrincipal, cfgFunc ClusterClientConfigFunc, cluster Cluster) (client.Client, error) {
+	if user == nil {
+		user = &auth.UserPrincipal{
+			ID: "weave-gitops-server",
```

Comment:
Why do we need to set this? Can we not return an error instead? I'm thinking this _may_ be lead to privilege escalation if there is RBAC in place that refers to this "user".